src/hotspot/cpu/aarch64/matcher_aarch64.hpp

```diff
@@ -164,10 +164,14 @@
   // Implements a variant of EncodeISOArrayNode that encode ASCII only
   static const bool supports_encode_ascii_array = true;
 
-  // SVE requires vector indices for gather-load/scatter-store operations
-  // on all data types.
-  static bool gather_scatter_needs_vector_index(BasicType bt) {
-    return has_predicated_vectors();
+  // Return true if this CPU requires the index input saved in an array address
+  // for vector gather-load/scatter-store operations. Otherwise, return false if
+  // the index input should be saved in a vector register.
+  //
+  // SVE requires vector indices for gather-load/scatter-store operations on all
+  // data types.
+  static bool gather_scatter_requires_index_in_address(BasicType bt) {
```

Comment:
I know I agreed to this naming, but I looked at the signature of `Gather` again:
`LoadVectorGatherNode(Node* c, Node* mem, Node* adr, const TypePtr* at, const TypeVect* vt, Node* indices)`

I'm a little confused now what is the `address` that your name references. Is it the `adr`? I think not, because that is the base address, right? Can you clarify a little more? Maybe add to the documentation of the gather and scatter node as well, if you think that helps?

Reply:
Ok, `gather_scatter_requires_indices_from_array` sounds better to me. I will change it soon.

>I'm a little confused now what is the address that your name references. Is it the adr? I think not, because that is the base address, right? Can you clarify a little more? Maybe add to the documentation of the gather and scatter node as well, if you think that helps?

It means the input `indices` is an address that saves the indexes if this method return true, otherwise, `indices` is a vector register. You are right that it has no relationship with `adr` input which is the memory base address.